python/tank/platform/engine.py

```diff
@@ -979,8 +984,11 @@ def _do_invoke(self):
                         Execute function and return result
                         """
                         self._res = self._fn()
-                        
-                return Invoker()
+
+                # Make sure that the invoker is for the main thread only.
+                invoker = Invoker()
+                invoker.moveToThread(QtCore.QThread.currentThread())
```

Comment:
Yes, it was a bug. Since I am creating the invoker on the fly, it can be created in a background thread.

In Toolkit, we should also add the moveToThread call I think so that if the engine is instantiated in a background thread for some reason it won't lock up.